src/main/java/net/minecraftforge/fml/client/config/GuiUtils.java

```diff
@@ -213,10 +238,32 @@ public static void drawTexturedModalRect(int x, int y, int u, int v, int width,
      *                     Set to a negative number to have no max width.
      * @param font the font for drawing the text in the tooltip box
      */
-    public static void drawHoveringText(List<String> textLines, final int mouseX, final int mouseY, final int screenWidth, final int screenHeight, final int maxTextWidth, FontRenderer font)
+    public static void drawHoveringText(List<String> textLines, int mouseX, int mouseY, int screenWidth, int screenHeight, int maxTextWidth, FontRenderer font)
+    {
+        drawHoveringText(cachedTooltipStack, textLines, mouseX, mouseY, screenWidth, screenHeight, maxTextWidth, font);
+    }
+    
+    /**
+     * Use this version if calling from somewhere where ItemStack context is available.
+     * 
+     * @see #drawHoveringText(List, int, int, int, int, int, FontRenderer)
+     */
+    public static void drawHoveringText(@Nullable final ItemStack stack, List<String> textLines, int mouseX, int mouseY, int screenWidth, int screenHeight,
+            int maxTextWidth, FontRenderer font)
     {
         if (!textLines.isEmpty())
         {
+            RenderTooltipEvent.Pre event = new RenderTooltipEvent.Pre(stack, textLines, mouseX, mouseY, screenWidth, screenHeight, maxTextWidth, font);
```

Comment:
Well, my thinking was, you keep the check as is, but move the "Pre" event outside the check. If someone adds some lines, the check passes, and it renders. If it doesn't pass (no event fired, or no lines were added), then no harm and no tooltip.